packages/relay-compiler/codegen/RelayFileWriter.js

```diff
@@ -353,13 +360,39 @@ class RelayFileWriter implements FileWriterInterface {
       return allOutputDirectories;
     });
   }
-}
 
-function md5(x: string): string {
-  return crypto
-    .createHash('md5')
-    .update(x, 'utf8')
-    .digest('hex');
+  /**
+   * Find all *.queryMap.json and write it into a single file.
+   * @param allOutputDirectories
+   */
+  writeCompleteQueryMap(
+    allOutputDirectories: Map<string, CodegenDirectory>,
+  ): void {
+    const queryMapFilePath = `${this._config.baseDir}/queryMap.json`;
+    try {
+      let queryMapJson = {};
+      allOutputDirectories.forEach(d => {
+        fs.readdirSync(d._dir).forEach(f => {
+          if (f.endsWith('queryMap.json')) {
+            const singleQueryMap = JSON.parse(
+              fs.readFileSync(path.join(d._dir, f), 'utf8'),
+            );
+            queryMapJson = {
+              ...queryMapJson,
+              ...singleQueryMap,
+            };
+          }
+        });
+      });
+
+      fs.writeFileSync(queryMapFilePath, JSON.stringify(queryMapJson, null, 2));
+      this._reporter.reportMessage(
+        `Written complete queryMap file to ${queryMapFilePath}`,
```

Comment:
Grammar nit: "Wrote"
`Wrote complete queryMap file to ${queryMapFilePath}`

Or, word order change:
`Complete queryMap written to file $${queryMapFilePath}`

Reply:
Thanks! Done.